### ISSUE TITLE
State based hashcode for value types

### ISSUE THREAD

**Issue by hangshao0** [eclipse-openj9/openj9#15529]
The hash code of a value type should be based on its state rather than the identity (address). 
Currently OpenJ9 still calculates the hash code of value types using its address rather than its value.

**Comment by hangshao0** [comment]
This is related to https://github.com/eclipse-openj9/openj9/pull/15497.

JCL has a private helper in class `PrimitiveObjectMethods` that calculates the VT hash code:
`
 private static int primitiveObjectHashCode(Object o)`

**Comment by hangshao0** [comment]
@ehrenjulzert will be look at this.

**Comment by hangshao0** [comment]
Doing call-in inside existing native method `convertValueToHash()` is not an option, see the discussions here: https://github.com/eclipse-openj9/openj9/pull/15741.

**Comment by ehrenjulzert** [comment]
As we discussed in #15880, implementing hashcodes for value types natively will be difficult and require changes to the GC. It also may not be worth it to sink too too much time into the native implementation since the spec may change in the future to remove the need to hash value types at all. For this reason I've decided to instead make a temporary implementation purely in Java which just calls the `primitiveObjectHashCode` helper function.
